server/paymentProviders/opencollective/collective.js

```diff
@@ -87,10 +87,23 @@ paymentMethodProvider.processOrder = async (order, options = {}) => {
     if (order.collective.HostCollectiveId !== order.paymentMethod.CollectiveId) {
       throw new Error(`You need to use the payment method of the host (${order.collective.HostCollectiveId}) to add funds to this collective`);
     }
-
-    // If Hosts are not the same, then look for fromCollectiveHost Credit Card
+    // If Hosts are not the same, then check if both have the same currency collectives
+    // and also both hosts of these collectives have the same currency as well
+    // then look for fromCollectiveHost Credit Card
     // and create transaction through the paymentLib Process order
   } else if (fromCollectiveHost.id !== collectiveHost.id) {
+    const fromCollectiveHost = await order.fromCollective.getHostCollective();
+    const collectiveHost = await order.collective.getHostCollective();
+    // Check if collectives have the same currency
+    if (order.fromCollective.currency !== order.collective.currency) {
+      throw new Error(`Payments Across hosts are only allowed when both Collectives have the same currency. Collective ${order.collective.name}` +
+        ` is ${order.collective.currency} and ${order.fromCollective.name} is ${order.fromCollective.currency}.`);
+    }
+    // Check if Hosts have the same currency as well
+    if (fromCollectiveHost.currency !== collectiveHost.currency) {
```

Comment:
running the following query in staging will return 4 rows: `select * from "Transactions" where "description"='Donation to WWCode East Bay' order by id Desc;`
the 2 lasts are the one generated by me using the UI in staging:
`Open Collective Company(currency USD host open Collective USD) donates 10USD to WWCode East Bay(currency USD host WWcode USD)` .
All other cases fail(multi currency)

Reply:
In all of those cases, the currency of the collectives and the hosts are in USD.

I wanted to see examples where `fromCollective.currency` === `toCollective.currency` but `fromHostCollective.currency` !== `toHostCollective.currency` or `fromHostCollective.currency` !== `fromCollective.currency`.

An example of that would be `/brusselstogether` (EUR:EUR) giving money to `/wwcodeberlin` (EUR:USD).

Reply:
I've already tried them and they failed as they should because the hosts have different currencies: getting `Payment Across Hosts are only allowed when both Hosts have the same currency`

Reply:
Tried paying WWCodeBarcelona from OpenCollective-company

![screen shot 2018-08-27 at 5 33 24 pm](https://user-images.githubusercontent.com/8717041/44687501-6204f400-aa1f-11e8-8c25-b6e3fe565bca.png)

Reply:
the example you asked: 

![screen shot 2018-08-27 at 5 37 01 pm](https://user-images.githubusercontent.com/8717041/44687694-db044b80-aa1f-11e8-9f33-58c59d6eb92e.png)

Reply:
My initial comment (https://github.com/opencollective/opencollective-api/pull/1416#discussion_r213102366) was about removing that error for those cases because I don't think that they are problematic. Can you remove this error message and try those cases and share the results in the `Transactions` table?

Reply:
of course, will check those as well!